src/EventStore.Core/TransactionLog/Scavenging/Sqlite/SqliteScavengeBackend.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using EventStore.Common.Log;
+using Microsoft.Data.Sqlite;
+
+namespace EventStore.Core.TransactionLog.Scavenging.Sqlite {
+	public class SqliteScavengeBackend {
+		protected static readonly ILogger Log = LogManager.GetLoggerFor<SqliteScavengeBackend>();
+	}
+
+	// Encapsulates a connection to sqlite, complete with prepared statements and an API to access it
+	public class SqliteScavengeBackend<TStreamId> : SqliteScavengeBackend, IScavengeStateBackend<TStreamId> {
+		// WAL with SYNCHRONOUS NORMAL means that
+		//  - commiting a transaction does not wait to it to flush to disk
+		//  - which is nice and quick, but means in powerloss the last x transactions
+		//    can be lost. the database will be in a valid state.
+		//  - this is suitable for us because scavenge will continue from the last
+		//    persisted checkpoint.
+		private const string SqliteWalJournalMode = "wal";
+		private const int SqliteNormalSynchronousValue = 1;
+		private const int DefaultSqliteCacheSize = 2 * 1024 * 1024;
+		private const int SqlitePageSize = 16 * 1024;
+		private readonly int _cacheSizeInBytes;
+		private SqliteBackend _sqliteBackend;
+
+		private const int SchemaVersion = 1;
+		
+		public IScavengeMap<TStreamId, Unit> CollisionStorage { get; private set; }
+		public IScavengeMap<ulong,TStreamId> Hashes { get; private set; }
+		public IMetastreamScavengeMap<ulong> MetaStorage { get; private set; }
+		public IMetastreamScavengeMap<TStreamId> MetaCollisionStorage { get; private set; }
+		public IOriginalStreamScavengeMap<ulong> OriginalStorage { get; private set; }
+		public IOriginalStreamScavengeMap<TStreamId> OriginalCollisionStorage { get; private set; }
+		public IScavengeMap<Unit,ScavengeCheckpoint> CheckpointStorage { get; private set; }
+		public IScavengeMap<int,ChunkTimeStampRange> ChunkTimeStampRanges { get; private set; }
+		public IChunkWeightScavengeMap ChunkWeights { get; private set; }
+		public ITransactionFactory<SqliteTransaction> TransactionFactory { get; private set; }
+		public ITransactionManager TransactionManager { get; private set; }
+
+		public SqliteScavengeBackend(int cacheSizeInBytes = DefaultSqliteCacheSize) {
+			_cacheSizeInBytes = Math.Abs(cacheSizeInBytes);
```

Comment:
is there any specific reason why we automatically fix a negative cache size to a positive value here?
for me, it seems counter-intuitive and an exception seems to be more in line with how other config params work

Reply:
i don't think so, ive removed the abs, and being negative is validated in the node settings